commit 31367baf86c73478b6c8aa9b4453fbf9decbfdc4
Author: Mrokkk <mrokk@protonmail.com>
Date:   Sat Jun 1 16:46:57 2024 +0200

    Add support for Phoenix
    
    * disable development mode
    * both x86_64 and i386 support added

diff --git a/bfd/config.bfd b/bfd/config.bfd
index bbf12447..1c6fcd65 100644
--- a/bfd/config.bfd
+++ b/bfd/config.bfd
@@ -621,6 +621,10 @@ case "${targ}" in
     targ_defvec=i386_coff_go32_vec
     targ_selvecs="i386_coff_go32stubbed_vec i386_aout_vec"
     ;;
+  i[3-7]86-*-phoenix*)
+    targ_defvec=i386_elf32_vec
+    targ_selvecs="iamcu_elf32_vec"
+    ;;
   i[3-7]86-*-darwin* | i[3-7]86-*-macos10* | i[3-7]86-*-rhapsody*)
     targ_defvec=i386_mach_o_vec
     targ_selvecs="mach_o_le_vec mach_o_be_vec mach_o_fat_vec pef_vec pef_xlib_vec sym_vec"
@@ -721,6 +725,11 @@ case "${targ}" in
     want64=true
     targ_underscore=no
     ;;
+  x86_64-*-phoenix*)
+    targ_defvec=x86_64_elf64_vec
+    targ_selvecs="i386_elf32_vec"
+    want64=true
+    ;;
   x86_64-*-rdos*)
     targ_defvec=x86_64_elf64_vec
     want64=true
diff --git a/bfd/development.sh b/bfd/development.sh
index 1112d2ec..535d7406 100644
--- a/bfd/development.sh
+++ b/bfd/development.sh
@@ -16,7 +16,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 # Controls whether to enable development-mode features by default.
-development=true
+development=false
 
 # Indicate whether this is a release branch.
 experimental=false
diff --git a/config.sub b/config.sub
index defe52c0..39a66f25 100755
--- a/config.sub
+++ b/config.sub
@@ -1766,7 +1766,7 @@ case $os in
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* | cos* | mbr* )
+	     | fiwix* | mlibc* | cos* | mbr* | phoenix* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff --git a/gas/configure.tgt b/gas/configure.tgt
index 7c66734e..81e65f83 100644
--- a/gas/configure.tgt
+++ b/gas/configure.tgt
@@ -237,6 +237,7 @@ case ${generic_target} in
   i386-*-knetbsd*-gnu | \
   i386-*-netbsd* | \
   i386-*-openbsd*)			fmt=elf em=nbsd ;;
+  i386-*-phoenix*)			fmt=elf em=linux ;;
   i386-*-linux-*)			fmt=elf em=linux
     case ${cpu}-${os} in
       x86_64*-linux-gnux32)		arch=x86_64:32 ;;
diff --git a/ld/Makefile.am b/ld/Makefile.am
index f9ee05b1..8de4b04e 100644
--- a/ld/Makefile.am
+++ b/ld/Makefile.am
@@ -372,7 +372,8 @@ ALL_EMULATION_SOURCES = \
 	exgateelf.c \
 	ez80.c \
 	ez8001.c \
-	ez8002.c
+	ez8002.c \
+	eelf_i386_phoenix.c
 
 ALL_EMULATIONS = $(ALL_EMULATION_SOURCES:.c=.@OBJEXT@)
 
@@ -467,7 +468,8 @@ ALL_64_EMULATION_SOURCES = \
 	eelf_x86_64_sol2.c \
 	ehppa64linux.c \
 	ei386pep.c \
-	emmo.c
+	emmo.c \
+	eelf_x86_64_phoenix.c
 
 ALL_64_EMULATIONS = $(ALL_64_EMULATION_SOURCES:.c=.@OBJEXT@)
 
diff --git a/ld/configure.tgt b/ld/configure.tgt
index f937f78b..fe84f524 100644
--- a/ld/configure.tgt
+++ b/ld/configure.tgt
@@ -446,6 +446,9 @@ i[3-7]86-*-beos*)	targ_emul=elf_i386_be
 			;;
 i[3-7]86-*-haiku*)	targ_emul=elf_i386_haiku
 			;;
+i[3-7]86-*-phoenix*)	targ_emul=elf_i386_phoenix
+			targ_extra_emuls=elf_i386
+			;;
 i[3-7]86-*-vxworks*)	targ_emul=elf_i386_vxworks
 			;;
 i[3-7]86-*-chaos)	targ_emul=elf_i386_chaos
@@ -1020,6 +1023,9 @@ x86_64-*-linux-*)	targ_emul=elf_x86_64
 			targ_extra_libpath="elf_i386 elf32_x86_64"
 			tdir_elf_i386=`echo ${targ_alias} | sed -e 's/x86_64/i386/'`
 			;;
+x86_64-*-phoenix*)	targ_emul=elf_x86_64_phoenix
+			targ_extra_emuls="elf_x86_64 elf_i386"
+			;;
 x86_64-*-redox*)	targ_emul=elf_x86_64
 			targ_extra_emuls=elf_i386
 			;;
diff --git a/ld/emulparams/elf_i386_phoenix.sh b/ld/emulparams/elf_i386_phoenix.sh
new file mode 100644
index 00000000..4ba9e178
--- /dev/null
+++ b/ld/emulparams/elf_i386_phoenix.sh
@@ -0,0 +1,2 @@
+source_sh ${srcdir}/emulparams/elf_i386.sh
+TEXT_START_ADDR=0x1000
diff --git a/ld/emulparams/elf_phoenix.sh b/ld/emulparams/elf_phoenix.sh
new file mode 100644
index 00000000..59e8e04e
--- /dev/null
+++ b/ld/emulparams/elf_phoenix.sh
@@ -0,0 +1 @@
+source_sh ${srcdir}/emulparams/elf_i386_phoenix.sh
diff --git a/ld/emulparams/elf_x86_64_phoenix.sh b/ld/emulparams/elf_x86_64_phoenix.sh
new file mode 100644
index 00000000..59e7df71
--- /dev/null
+++ b/ld/emulparams/elf_x86_64_phoenix.sh
@@ -0,0 +1 @@
+source_sh ${srcdir}/emulparams/elf_x86_64.sh
diff --git a/ld/emulparams/i386phoenix.sh b/ld/emulparams/i386phoenix.sh
new file mode 100644
index 00000000..59e8e04e
--- /dev/null
+++ b/ld/emulparams/i386phoenix.sh
@@ -0,0 +1 @@
+source_sh ${srcdir}/emulparams/elf_i386_phoenix.sh
